src/v/kafka/server/group.cc

```diff
@@ -70,7 +75,13 @@ group::group(
   , _num_members_joining(0)
   , _new_member_added(false)
   , _conf(conf)
-  , _partition(std::move(partition)) {
+  , _partition(std::move(partition))
+  , _recovery_policy(
+      config::shard_local_cfg().rm_violation_recovery_policy.value()) {
+    if (_recovery_policy != crash && _recovery_policy != best_effort) {
+        vassert(false, "Unknown recovery policy: {}", _recovery_policy);
+    }
```

Comment:
related to the comment below about `enum class`. switching to enum class will mean that you have strong typing so this check may be unnecessary: the compiler will enforce that the types are crash or best effort since those are the only two options.